### PR TITLE
Fixes code broken by Bee's changes (and my own negligence)

### PR DESCRIPTION
**Pull request by MaltVinegar**
Going to be honest, I haven't even tested this stuff yet

## Changelog
:cl:
refactor: deepcore & cryo's update_icon_state calls parent function
refactor: cryo uses new emag system
tweak: resolved a merge conflict in the config that I somehow missed
github: allows map greps to ignore directional firelocks, credits to Ku beRute
/:cl:

### CODE REVIEW

File: singulostation/code/game/machinery/autodoc.dm [hunk 1]
```diff
@@ -161,8 +161,10 @@
...
-/obj/machinery/autodoc/emag_act(mob/user)
-	if(obj_flags & EMAGGED)
-		return
-	obj_flags |= EMAGGED
+/obj/machinery/autodoc/should_emag(mob/user)
+	return ..()
+
+/obj/machinery/autodoc/on_emag(mob/user)
+	..()
+
```
Comment (KubeRoot) [inline singulostation/code/game/machinery/autodoc.dm]:
```suggestion
```

Aren't those redundant?

Comment (MaltVinegar) [reply singulostation/code/game/machinery/autodoc.dm]: You right

File: singulostation/code/game/machinery/autodoc.dm [hunk 2]
```diff
@@ -161,4 +161,7 @@
 	if(panel_open)
 		overlays += "[icon_state]_panel"
 
+/obj/machinery/autodoc/on_emag(mob/user)
+	..()
+
```
Comment (KubeRoot) [inline singulostation/code/game/machinery/autodoc.dm:167]: Isn't it redundant though? Or do you need to prevent it from returning?

Comment (KubeRoot) [reply singulostation/code/game/machinery/autodoc.dm:167]: Ah, I see, the empty line with no indentation confused me, carry on

Comment (MaltVinegar) [reply singulostation/code/game/machinery/autodoc.dm:167]: It needs to call the parent, and this is how it's done everywhere else I checked
See: #223

Comment (MaltVinegar) [reply singulostation/code/game/machinery/autodoc.dm:167]: Though, now that I think about it, should_emag should probably return true

Comment (MaltVinegar) [reply singulostation/code/game/machinery/autodoc.dm:167]: Ah, looks like I need to se `emag_toggleable` to true

**Pull request merged**